conda/core/index.py

```diff
@@ -328,7 +332,65 @@ def maybe_decompress(filename, resp_content):
                              getattr(e.response, 'elapsed', None))
 
 
-def read_local_repodata(cache_path):
+def write_pickled_repodata(cache_path, repodata):
+    # Don't bother to pickle empty channels
+    if not repodata.get('packages'):
+        return
+    try:
+        with open(cache_path + '.q', 'wb') as f:
+            pickle.dump(repodata, f)
+    except Exception as e:
+        import traceback
+        log.debug("Failed to dump pickled repodata.\n%s", traceback.format_exc())
+
+
+def read_pickled_repodata(cache_path, channel_url, schannel, priority):
+    pickle_path = cache_path + '.q'
+    # Don't trust pickled data if there is no accompanying json data
+    if not isfile(pickle_path) or not isfile(cache_path):
+        return None
+    try:
+        with open(cache_path + '.q', 'rb') as f:
+            repodata = pickle.load(f)
+    except Exception as e:
+        import traceback
+        log.debug("Failed to load pickled repodata.\n%s", traceback.format_exc())
+        return None
+    if repodata.get('_url') != channel_url:
+        return None
+    add_pip = context.add_pip_as_python_dependency
+    changed = False
+    # Safe changes to IndexRecord
+    if ((repodata['_add_pip'] != add_pip or
+         repodata['_priority'] != priority or
+         repodata['_schannel'] != schannel)):
+        repodata['_priority'] = priority
+        repodata['_add_pip'] = add_pip
+        changed = True
+        for rec in itervalues(repodata.get('packages', {})):
+            rec.priority = priority
```

Comment:
If this loop is a hot spot, we can make `rec.priority` an object with a `__get__` method, so that we can reference it once for the whole repodata set.  And then a single operation can switch the whole group.

Something like

```
class Priority(object):
    __slots__ = ('_priority',)
    def __init__(self, priority):
        self._priority = priority
    def __get__(self):
        return self._priority
```

Then in `process_repodata` line 422 could become

    repodata['_priority'] = priority = Priority(priority)

and instead of the for-loop here, we just have

    repodata['_priority']._priority = priority